ecomap/ecomap/RootViewControllers/MapViewController.m

```diff
@@ -0,0 +1,307 @@
+//
+//  ViewController.m
+//  ecomap
+//
+//  Created by Anton Kovernik on 02.02.15.
+//  Copyright (c) 2015 SoftServe. All rights reserved.
+//
+
+#import "MapViewController.h"
+#import "EcomapRevealViewController.h"
+#import <GoogleMaps/GoogleMaps.h>
+#import "EcomapProblem.h"
+#import "EcomapFetcher.h"
+#import "Spot.h"
+#import "NonHierarchicalDistanceBasedAlgorithm.h"
+#import "GDefaultClusterRenderer.h"
+#import "EcomapClusterRenderer.h"
+#import "ProblemViewController.h"
+#import "EcomapProblemFilteringMask.h"
+#import "EcomapFilter.h"
+#import "GlobalLoggerLevel.h"
+
+#define SOCKET_ADDRESS @"http://176.36.11.25:8091"
+#define FILTER_ON NO
+
+@interface MapViewController () <CLLocationManagerDelegate>
+
+@property (weak, nonatomic) IBOutlet UIBarButtonItem *revealButtonItem;
+@property (nonatomic, strong) GClusterManager *clusterManager;
+@property (nonatomic, strong) CLLocationManager *locationManager;
+@property (nonatomic, strong) GMSMapView *mapView;
+@property (nonatomic, strong) NSSet *markers;
+@property (nonatomic, strong) GMSCameraPosition *previousCameraPosition;
+@property (nonatomic, strong) NSSet *problems;
+@property (nonatomic, strong) SRWebSocket *socket;
+
+@end
+
+@implementation MapViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self customSetup];
+    [self mapSetup];
+    [self socketInit];
+}
+
+- (void)socketInit {
```

Comment:
We have decided to use sockets to notify clients that there is new data available (new problem posted) and clients should renew it.